packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -91,37 +91,34 @@ export default function UnsavedInnerBlocks( {
 		}
 	);
 
-	const { isSaving, draftNavigationMenus, hasResolvedDraftNavigationMenus } =
-		useSelect(
-			( select ) => {
-				if ( isDisabled ) {
-					return EMPTY_OBJECT;
-				}
-
-				const {
-					getEntityRecords,
-					hasFinishedResolution,
-					isSavingEntityRecord,
-				} = select( coreStore );
-
-				return {
-					isSaving: isSavingEntityRecord(
-						'postType',
-						'wp_navigation'
-					),
-					draftNavigationMenus: getEntityRecords(
-						...DRAFT_MENU_PARAMS
-					),
-					hasResolvedDraftNavigationMenus: hasFinishedResolution(
-						'getEntityRecords',
-						DRAFT_MENU_PARAMS
-					),
-				};
-			},
-			[ isDisabled ]
-		);
-
-	const { hasResolvedNavigationMenus, navigationMenus } = useNavigationMenu();
+	const { isSaving, hasResolvedDraftNavigationMenus } = useSelect(
+		( select ) => {
+			if ( isDisabled ) {
+				return EMPTY_OBJECT;
+			}
+
+			const {
+				getEntityRecords,
+				hasFinishedResolution,
+				isSavingEntityRecord,
+			} = select( coreStore );
+
+			return {
+				isSaving: isSavingEntityRecord( 'postType', 'wp_navigation' ),
+				draftNavigationMenus: getEntityRecords(
+					// This is needed so that hasResolvedDraftNavigationMenus gives the correct status.
```

Comment:
We have to keep this line, otherwise hasResolvedDraftNavigationMenus doesn't work.

Reply:
Why?

Reply:
Can we just not check if navigation menus are resolved? We only need them because
```
Also ensure other navigation menus have loaded so an
		// accurate name can be created.
```
Is there another way? Like wait for this when creating the default title instead?

Reply:
I don't think so actually - we also need to check if draft navigations are created because navigation we create is a draft. If I remove the check for `hasResolvedDraftNavigationMenus` then two get created.

Reply:
When we call createNavigationMenu we do it with an empty title and there is a hook `useGenerateDefaultNavigationTitle` where we alreadt check that we loaded all drafts already so we can assign the correct number.

```const [ draftNavigationMenus, navigationMenus ] = await Promise.all( [
			getEntityRecords( ...DRAFT_MENU_PARAMS ),
			getEntityRecords( ...PUBLISHED_MENU_PARAMS ),
		] );
```

That means this guard here is kinda pointless.

Reply:
That doesn't stop us calling `createNavigationMenu` twice though does it?

Reply:
Why don't you pass in the status of the `createNavigationMenu` from the `edit` component? That way we have a way of telling whether the conversion is in progress. That might help us block the 2nd menu because it gives a true status of the progress of the event we care about - namely: whether we are already creating a navigation.

Relying on various resolved states is surely prone to error.

Reply:
> That doesn't stop us calling createNavigationMenu twice though does it?

I didn't mean that. I mean that `hasResolvedDraftNavigationMenus` and `hasResolvedNavigationMenus` are only needed here as deps because we check for them in the effect - but `createNavigationMenu` which is the only result of the effect ALSO checks them - it seems they're not needed. And if they're not needed we don't need `draftNavigationMenus` either.

Reply:
If I remove them then the problem comes back, so I think they are needed.

Reply:
> Why don't you pass in the isCreatingNavigationMenu status of the createNavigationMenu from the edit component?

I don't think this will work. After the new navigation is created, the status is reset to `idle`,  but the block is still dirty as the new (draft) navigation hasn't yet loaded, so the effect gets triggered again, and we end up in a loop creating endless new menus.

Reply:
> I didn't mean that. I mean that hasResolvedDraftNavigationMenus and hasResolvedNavigationMenus are only needed here as deps because we check for them in the effect - but createNavigationMenu which is the only result of the effect ALSO checks them - it seems they're not needed. And if they're not needed we don't need draftNavigationMenus either.

I don't think this will work, because although `createNavigationMenu` does check the status of those dependencies, it happens after we call `createNavigationMenu`, so the useEffect that is checking whether the blocks are in a dirty state will continue to be called.